### PR TITLE
BUG warn on dangling symlinks

### PR DESCRIPTION
**Pull request by beckermr**
Closes #3844 

@isuruf for viz

### CODE REVIEW

File: conda_build/build.py [hunk 1]
```diff
@@ -1280,6 +1282,7 @@ def build_info_files_json_v1(m, prefix, files, files_with_prefix):
         elif not isfile(path):
             # this is a softlink that points to nowhere, so is zero bytes
             file_info["size_in_bytes"] = 0
+            log.warn('file %s is a symlink with no target', path)
```
Comment (marcelotrevisani) [inline conda_build/build.py]:
```suggestion
            warnings.warn('file %s is a symlink with no target', path)
```
should be fine

Comment (marcelotrevisani) [reply conda_build/build.py]: you will need to import warnings as well
```python
import warnings
```

**Comment by beckermr** [comment]
Done @marcelotrevisani !

**Comment by marcelotrevisani** [comment]
Thanks again for your contribution! :)

**Pull request merged**